src/stringImproved.h

```diff
@@ -146,7 +151,7 @@ class string : public std::string
     }
     bool endswith(char suffix) const
     {
-        return !empty() && (*this)[length()-1] == suffix;
```

Comment:
This is interesting, because it meant that `string{}.endswith("") != string{}.endswith('\0')` (the string version assumes an empty string is always present at the end of a string)